src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -51,11 +57,70 @@ public SecretManager(ISecretsRepository repository, IKeyValueConverterFactory ke
             _metricsLogger = metricsLogger ?? throw new ArgumentNullException(nameof(metricsLogger));
             _repositoryClassName = _repository.GetType().Name.ToLower();
             _hostNameProvider = hostNameProvider;
+            _startupContextProvider = startupContextProvider;
+        }
 
-            if (createHostSecretsIfMissing)
+        private ConcurrentDictionary<string, IDictionary<string, string>> FunctionSecrets
```

Comment:
In the latest iteration, I moved the cache init out of the ctor and made it on demand, only when secrets are requested